src/main/scala/firrtl/LoweringCompilers.scala

```diff
@@ -97,7 +97,6 @@ class LowFirrtlOptimization extends CoreTransform {
   def outputForm = LowForm
   def transforms = Seq(
     passes.RemoveValidIf,
-    passes.ConstProp,
     passes.PadWidths,
```

Comment:
@azidar Do you know why ConstProp was running before PadWidths? This ordering goes waaaay back. I think there is still a problem with this PR in rocket-chip but deleting this first run of ConstProp fixes the following bug:
```
wire w : { a : UInt<4>, b : UInt<4> }
w.a <= UInt<2>(3)
w.b <= UInt<2>(3)
node x = cat(w.a, w.b)
```
Without constant prop across wires this was fine, but with these changes ConstProp turns it into:
```
node x = UInt<4>("hf")
```
We need to run PadWidths before constant propagating such a concatenation.

An alternative I'm also trying out is padding connections to wires in ConstProp since there seem to be problems with not running constprop before padwidths